pycode/memilio-surrogatemodel/memilio/surrogatemodel/ode_secir_groups/data_generation.py

```diff
@@ -0,0 +1,302 @@
+        data['damping_day'].append(damping_day)
+        bar.next()
+    bar.finish()
+
+    if normalize:
+        # logarithmic normalization
+        transformer = FunctionTransformer(np.log1p, validate=True)
+        inputs = np.asarray(data['inputs']).transpose(2, 0, 1).reshape(48, -1)
+        scaled_inputs = transformer.transform(inputs)
+        scaled_inputs = scaled_inputs.transpose().reshape(num_runs, input_width, 48)
+        scaled_inputs_list = scaled_inputs.tolist()
+
+        labels = np.asarray(data['labels']).transpose(2, 0, 1).reshape(48, -1)
+        scaled_labels = transformer.transform(labels)
+        scaled_labels = scaled_labels.transpose().reshape(num_runs, label_width, 48)
+        scaled_labels_list = scaled_labels.tolist()
+
+        # cast dfs to tensors
+        data['inputs'] = tf.stack(scaled_inputs_list)
+        data['labels'] = tf.stack(scaled_labels_list)
+
+    if save_data:
+        # check if data directory exists. If necessary, create it.
+        if not os.path.isdir(path_out):
+            os.mkdir(path_out)
+
+        # save dict to json file
+        with open(os.path.join(path_out, 'data_secir_groups.pickle'), 'wb') as f:
+            pickle.dump(data, f)
+    return data
+
+
+def getBaselineMatrix():
+    """! loads the baselinematrix
+    """
+
+    baseline_contact_matrix0 = os.path.join(
+        "./data/contacts/baseline_home.txt")
+    baseline_contact_matrix1 = os.path.join(
+        "./data/contacts/baseline_school_pf_eig.txt")
+    baseline_contact_matrix2 = os.path.join(
+        "./data/contacts/baseline_work.txt")
+    baseline_contact_matrix3 = os.path.join(
+        "./data/contacts/baseline_other.txt")
+
+    baseline = np.loadtxt(baseline_contact_matrix0) \
+        + np.loadtxt(baseline_contact_matrix1) + \
+        np.loadtxt(baseline_contact_matrix2) + \
+        np.loadtxt(baseline_contact_matrix3)
+
+    return baseline
+
+
+def getMinimumMatrix():
+    """! loads the minimum matrix
+    """
+
+    minimum_contact_matrix0 = os.path.join(
+        "./data/contacts/minimum_home.txt")
+    minimum_contact_matrix1 = os.path.join(
+        "./data/contacts/minimum_school_pf_eig.txt")
+    minimum_contact_matrix2 = os.path.join(
+        "./data/contacts/minimum_work.txt")
+    minimum_contact_matrix3 = os.path.join(
+        "./data/contacts/minimum_other.txt")
+
+    minimum = np.loadtxt(minimum_contact_matrix0) \
+        + np.loadtxt(minimum_contact_matrix1) + \
+        np.loadtxt(minimum_contact_matrix2) + \
+        np.loadtxt(minimum_contact_matrix3)
+
+    return minimum
+
+
+def get_population(path):
+    """! read population data in list from dataset
+    @param path Path to the dataset containing the population data
+    """
+
+    with open(path) as f:
+        data = json.load(f)
+    population = []
+    for data_entry in data:
+        population_county = []
+        population_county.append(
+            data_entry['<3 years'] + data_entry['3-5 years'] / 2)
```

Comment:
Why is '3-5 years' divided by 2 and where does the rest of the '3-5' population go?